test/datasets.js

```diff
@@ -399,6 +376,109 @@ test('DatasetClient', function(datasetClient) {
     bulkFeatureUpdate.end();
   });
 
+  datasetClient.test('#listFeatures', function(listFeatures) {
+    var featureIds = [];
+
+    // Setup data
+    listFeatures.test('insert some', function(assert) {
+      var client = new MapboxClient(process.env.MapboxAccessToken);
+      assert.ok(client, 'created dataset client');
+      var features = randomFeatures(3).map(function(f) {
+        f.id = hat();
```

Comment:
@tristen not sure if this is the issue, but seems like if your ids are randomly generated, then you'll have a hard time making assertions about order in things like `start` and `reverse`. Maybe give the features ids 1, 2, 3 and see if the tests pass more consistently?